frontend/src/Editor/Components/Form/Form.jsx

```diff
@@ -0,0 +1,273 @@
+import React, { useRef, useState, useEffect } from 'react';
+import { SubCustomDragLayer } from '../../SubCustomDragLayer';
+import { SubContainer } from '../../SubContainer';
+// eslint-disable-next-line import/no-unresolved
+import { diff } from 'deep-object-diff';
+import _, { omit } from 'lodash';
+import { Box } from '../../Box';
+import { componentTypes } from '@/Editor/WidgetManager/components';
+import { generateUIComponents } from './FormUtils';
+
+export const Form = function Form(props) {
+  const {
+    id,
+    component,
+    width,
+    height,
+    containerProps,
+    removeComponent,
+    styles,
+    setExposedVariable,
+    darkMode,
+    currentState,
+    fireEvent,
+    properties,
+    registerAction,
+    resetComponent,
+    childComponents,
+    onEvent,
+    dataCy,
+    paramUpdated,
+  } = props;
+  const { visibility, disabledState, borderRadius, borderColor } = styles;
+  const { buttonToSubmit, loadingState, advanced, JSONSchema } = properties;
+  const backgroundColor =
+    ['#fff', '#ffffffff'].includes(styles.backgroundColor) && darkMode ? '#232E3C' : styles.backgroundColor;
+  const computedStyles = {
+    backgroundColor,
+    borderRadius: borderRadius ? parseFloat(borderRadius) : 0,
+    border: `1px solid ${borderColor}`,
+    height,
+    display: visibility ? 'flex' : 'none',
+    position: 'relative',
+    overflow: 'hidden auto',
+  };
+
+  const parentRef = useRef(null);
+  const childDataRef = useRef({});
+
+  const [childrenData, setChildrenData] = useState({});
+  const [isValid, setValidation] = useState(true);
+  const [uiComponents, setUIComponents] = useState([]);
+
+  registerAction('resetForm', async function () {
+    resetComponent();
+  });
+  registerAction(
+    'submitForm',
+    async function () {
+      if (isValid) {
+        onEvent('onSubmit', { component }).then(() => resetComponent());
+      } else {
+        fireEvent('onInvalid');
+      }
+    },
+    [isValid]
+  );
+
+  const extractData = (data) => {
+    const result = {};
+
+    for (const key in data) {
+      const item = data[key];
+
+      if (item.name === 'Text') {
+        const textKey = item?.keyValue ?? item?.text;
+        const nextItem = data[parseInt(key) + 1];
+
+        if (nextItem && nextItem.name !== 'Text') {
+          result[textKey] = { ...nextItem };
+          delete result[textKey].name;
+        }
+      }
+    }
+
+    return result;
+  };
+
+  useEffect(() => {
+    setUIComponents(generateUIComponents(JSONSchema, advanced, componentTypes));
```

Comment:
```suggestion
    setUIComponents(generateUIComponents(JSONSchema, advanced));
```

`componentTypes` can be imported directly in the utils file